### PR TITLE
Provide consistent examples of bindings for all SDKs

### PR DESCRIPTION
**Pull request by amulyavarote**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Modified bindings documentation page (under building blocks) for dapr.io

## Issue reference

This partially fulfills: #1969

### CODE REVIEW

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md [hunk 1]
```diff
@@ -9,19 +9,19 @@ weight: 300
 Output bindings enable you to invoke external resources without taking dependencies on special SDK or libraries.
 For a complete sample showing output bindings, visit this [link](https://github.com/dapr/quickstarts/tree/master/bindings).
 
-Watch this [video](https://www.youtube.com/watch?v=ysklxm81MTs&feature=youtu.be&t=1960) on how to use bi-directional output bindings.
+## Example:
 
-<div class="embed-responsive embed-responsive-16by9">
-<iframe width="560" height="315" src="https://www.youtube.com/embed/ysklxm81MTs?start=1960" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
-</div>
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources(output binding).
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md]:
```suggestion
The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources via an output binding.
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md [hunk 2]
```diff
@@ -9,19 +9,19 @@ weight: 300
...
+
+<img src="/images/building-block-bindings-example.png" width=1000 alt="Diagram showing bindings of example service">
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md]: this image has an extra thin grey line at the right hand side that needs to be cropped out.

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 3]
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
 
 For more info on bindings, read [this overview]({{<ref bindings-overview.md>}}).
 
-For a quickstart sample showing bindings, visit this [link](https://github.com/dapr/quickstarts/tree/master/bindings).
+## Example:
+
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The checkout service uses Dapr to trigger the application(input binding).
+
+<img src="/images/building-block-bindings-example.png" width=1000 alt="Diagram showing bindings of example service">
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]: this image has an extra thin grey line at the right hand side that needs to be cropped out.

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 4]
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
...
 
 ## 1. Create a binding
 
-An input binding represents an event resource that Dapr uses to read events from and push to your application.
+An output binding represents a resource that Dapr uses to invoke and send messages to.
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]: should this be describing an input biding instead?

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 5]
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
 
 For more info on bindings, read [this overview]({{<ref bindings-overview.md>}}).
 
-For a quickstart sample showing bindings, visit this [link](https://github.com/dapr/quickstarts/tree/master/bindings).
+## Example:
+
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The checkout service uses Dapr to trigger the application(input binding).
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]:
```suggestion
The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The checkout service uses Dapr to trigger the application via an input binding.
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 6]
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
...
 ```
 
-Here, you create a new binding component with the name of `myevent`.
+{{% /codetab %}}
 
-Inside the `metadata` section, configure the Kafka related properties such as the topics to listen on, the brokers and more.
+{{< /tabs >}}
 
-## 2. Listen for incoming events
+## 2. Listen for incoming events(Input Binding)
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]:
```suggestion
## 2. Listen for incoming events (input binding)
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 7]
```diff
@@ -18,79 +18,265 @@ Dapr bindings allow you to:
...
+
+{{< /tabs >}}
+
+### ACK-ing an event
+
+In order to tell Dapr that you successfully processed an event in your application, return a `200 OK` response from your HTTP handler.
+
+### Rejecting an event
+
+In order to tell Dapr that the event wasn't processed correctly in your application and schedule it for redelivery, return any response different from `200 OK`. For example, a `500 Error`.
```
Comment (willtsai) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]:
```suggestion
In order to tell Dapr that the event was not processed correctly in your application and schedule it for redelivery, return any response other than `200 OK`. For example, a `500 Error`.
```

**Review by willtsai — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md [hunk 8]
```diff
@@ -9,19 +9,19 @@ weight: 300
...
-</div>
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources via an output binding.
+
+<img src="/images/building-block-bindings-example.png" width=1000 alt="Diagram showing bindings of example service">
 
 ## 1. Create a binding
 
 An output binding represents a resource that Dapr uses to invoke and send messages to.
 
 For the purpose of this guide, you'll use a Kafka binding. You can find a list of the different binding specs [here]({{< ref setup-bindings >}}).
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md:22]:
```suggestion
For the purpose of this guide, you'll use a Kafka binding. You can find a list of all supported binding components [here]({{< ref setup-bindings >}}).
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 9]
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
...
+{{< /tabs >}}
 
-## 2. Listen for incoming events
+## 2. Listen for incoming events (input binding)
 
-Now configure your application to receive incoming events. If using HTTP, you need to listen on a `POST` endpoint with the name of the binding as specified in `metadata.name` in the file.  In this example, this is `myevent`.
+Now configure your application to receive incoming events. If using HTTP, you need to listen on a `POST` endpoint with the name of the binding as specified in `metadata.name` in the file. 
 
-*The following example shows how you would listen for the event in Node.js, but this is applicable to any programming language*
+Below are code examples that leverage Dapr SDKs for output binding.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]:
```suggestion
Below are code examples that leverage Dapr SDKs to demonstrate an output binding.
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 10]
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
...
+
+{{< /tabs >}}
+
+### ACK-ing an event
+
+In order to tell Dapr that you successfully processed an event in your application, return a `200 OK` response from your HTTP handler.
+
+### Rejecting an event
+
+In order to tell Dapr that the event was not processed correctly in your application and schedule it for redelivery, return any response other than `200 OK`. For example, a `500 Error`.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:284]: Do we really want to encourage returning a 500 error? Shouldn't this be handled gracefully?

Comment (amulyavarote) [reply daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md:284]: This part was present in the docs perviously. I can take it out if not needed.

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md [hunk 11]
```diff
@@ -9,19 +9,19 @@ weight: 300
 Output bindings enable you to invoke external resources without taking dependencies on special SDK or libraries.
 For a complete sample showing output bindings, visit this [link](https://github.com/dapr/quickstarts/tree/master/bindings).
 
-Watch this [video](https://www.youtube.com/watch?v=ysklxm81MTs&feature=youtu.be&t=1960) on how to use bi-directional output bindings.
+## Example:
 
-<div class="embed-responsive embed-responsive-16by9">
-<iframe width="560" height="315" src="https://www.youtube.com/embed/ysklxm81MTs?start=1960" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
-</div>
+The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources via an output binding.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md]:
```suggestion
The below code example loosely describes an application that processes orders. In the example, there is an order processing service which has a Dapr sidecar. The order processing service uses Dapr to invoke external resources, in this case a Kafka, via an output binding.
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md [hunk 12]
```diff
@@ -59,38 +67,271 @@ To deploy this into a Kubernetes cluster, fill in the `metadata` connection deta
...
 ```
 
 {{% /codetab %}}
 
 {{< /tabs >}}
 
-## 2. Send an event
+## 2. Send an event(Output Binding)
+
+Below are code examples that leverage Dapr SDKs for output binding.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-bindings.md]:
```suggestion
Below are code examples that leverage Dapr SDKs to demonstrate an output binding.
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 13]
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
...
+
+<img src="/images/building-block-input-binding-example.png" width=1000 alt="Diagram showing bindings of example service">
 
 ## 1. Create a binding
 
-An input binding represents an event resource that Dapr uses to read events from and push to your application.
+An input binding represents a resource that Dapr uses to read events from and push to your application.
 
-For the purpose of this HowTo, we'll use a Kafka binding. You can find a list of the different binding specs [here]({{< ref supported-bindings >}}).
+For the purpose of this guide, you'll use a Kafka binding. You can find a list of the different binding specs [here]({{< ref setup-bindings >}}).
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]:
```suggestion
For the purpose of this guide, you'll use a Kafka binding. You can find a full list of supported binding components [here]({{< ref setup-bindings >}}).
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 14]
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
...
 
-Create the following YAML file, named binding.yaml, and save this to a `components` sub-folder in your application directory.
-(Use the `--components-path` flag with `dapr run` to point to your custom components dir)
+Create a new binding component with the name of `checkout`.
+
+Inside the `metadata` section, configure Kafka related properties such as the topic to publish the message to and the broker.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]:
```suggestion
Inside the `metadata` section, configure Kafka related properties, such as the topic to publish the message to and the broker.
```

File: daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md [hunk 15]
```diff
@@ -18,79 +18,271 @@ Dapr bindings allow you to:
...
+
+{{< /tabs >}}
+
+### ACK-ing an event
+
+In order to tell Dapr that you successfully processed an event in your application, return a `200 OK` response from your HTTP handler.
+
+### Rejecting an event
+
+In order to tell Dapr that the event was not processed correctly in your application and schedule it for redelivery and handle it gracefully using logs or exceptions.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/bindings/howto-triggers.md]: Lets put this back to how it was until we discuss how best this should be handled. 

```suggestion
In order to tell Dapr that the event wasn't processed correctly in your application and schedule it for redelivery, return any response different from `200 OK`. For example, a `500 Error`.
```

**Comment by willtsai** [comment]
Looks like this comment from me is not yet addressed. Please either accept and commit the suggestion, or else resolve the comment if you are taking this in another direction.
https://github.com/dapr/docs/pull/2021/files#r762207024

**Review by willtsai — APPROVED** [review]

**Review by greenie-msft — APPROVED** [review]

**Pull request merged**